src/bluefinClient.ts

```diff
@@ -1974,4 +1977,34 @@ export class BluefinClient {
     }
     return response;
   };
+
+  signAndExecuteZkTransaction = async (tx: TransactionBlock) => {
+    try {
+      if (!this.signer || !this.walletAddress)
+        throw new Error("invalid signer or wallet address");
+      if (!this.getZkPayload()) throw new Error("invalid zk payloads");
```

Comment:
what is the purpose of this condition, as this condition can never ever be true. `this.getZkPayload()` will always return an object as per the current implementation of it.

Reply:
@abdullahabro yeah man you're right here. I can use isEmpty utility to verify that